docs/resources/rule_set.md

```diff
@@ -0,0 +1,138 @@
+
+  grant_rules {
+    principals = ["groups/${data.databricks_group.ds.display_name}"]
+    role = "roles/servicePrincipal.user"
+  }
+}
+```
+
+Rule set management through AWS Databricks account:
+
+```hcl
+// initialize provider at account-level
+provider "databricks" {
+  alias      = "mws"
+  host       = "https://accounts.cloud.databricks.com"
+  account_id = "00000000-0000-0000-0000-000000000000"
+  username   = var.databricks_account_username
+  password   = var.databricks_account_password
+}
+
+// account level group creation
+resource "databricks_group" "ds" {
+  display_name = "Data Science"
+}
+
+resource "databricks_service_principal" "automation_sp" {
+  display_name = "SP_FOR_AUTOMATION"
+}
+
+resource "databricks_rule_set" "automation_sp_rule_set" {
+  name = "accounts/${local.account_id}/servicePrincipals/${databricks_service_principal.automation_sp.application_id}/ruleSets/default"
+
+  grant_rules {
+    principals = ["groups/${databricks_group.ds.display_name}"]
+    role = "roles/servicePrincipal.user"
+  }
+}
+```
+
+Rule set management through Azure Databricks account:
+
+```hcl
+// initialize provider at Azure account-level
+provider "databricks" {
+  alias      = "azure_account"
+  host       = "https://accounts.azuredatabricks.net"
+  account_id = "00000000-0000-0000-0000-000000000000"
+  auth_type  = "azure-cli"
+}
+
+// account level group creation
+resource "databricks_group" "ds" {
+  display_name = "Data Science"
+}
+
+resource "databricks_service_principal" "automation_sp" {
+  display_name = "SP_FOR_AUTOMATION"
+}
+
+resource "databricks_rule_set" "automation_sp_rule_set" {
+  name = "accounts/${local.account_id}/servicePrincipals/${databricks_service_principal.automation_sp.application_id}/ruleSets/default"
+
+  grant_rules {
+    principals = ["groups/${databricks_group.ds.display_name}"]
+    role = "roles/servicePrincipal.user"
+  }
+}
+```
+
+## Argument Reference
+
+Rule set name and grant rules block argument are required.
+
+### Name Argument
+
+Unique identifier of a rule set. Currently, only default rule sets are supported. The following rule set formats are supported:
+* accounts/{account_id}/servicePrincipals/{service_principal_application_id}/ruleSets/default
+
+### Grant Rules Argument
+
+One or more `grant_rules` blocks are required to actually set access rules.
+
+```hcl
+grant_rules {
+  principals = [
+    "groups/{databricks_group.ds.display_name}"
+  ]
+  role = "roles/servicePrincipal.user"
+}
+```
+
+Arguments of the `grant_rules` block are:
+
+- `role` - (Required) Role to be granted. The following roles are supported:
+  * roles/servicePrincipal.manager - Manager of a service principal.
+  * roles/servicePrincipal.user - User of a service principal.
+- `principals` - (Required) a list of principals who are granted a role. The following format is supported:
+  * users/{username}
+  * groups/{groupname}
+  * servicePrincipals/{applicationId}
```

Comment:
Same here - can we allow to specify `users`, `service_principals`, `groups` arguments & gneerate `principals` from it?

Reply:
I am new to Go and terraform.. Do you have any tips for how to do that?

Reply:
@alexott Google provider does it as close to API as possible

https://registry.terraform.io/providers/hashicorp/google/4.22.0/docs/resources/api_gateway_api_iam